docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,213 @@
+* The `device_profile` command payload is an ordered list of modules representing any firmware, software and configuration combination.
+* The agent processes each module one by one in the same order, triggering sub-operations for the respective module type.
+* One successful installation of all the modules, the applied profile information is published to the same capability topic.
+* No rollback is performed on partial failures unless the subcommand for the failed module can rollback that single module.
+
+# Why device profile FAQ
+
+**Q1: Why do we need device profile when firmware, software and configuration management already exists**
+
+Performing and managing these operations individually for a long list of software and configuration items would be cumbersome,
+especially when performed on a large fleet of devices as each operation will have to be monitored and managed separately.
+Grouping them together into a single operation reduces that overhead considerably as you just have one operation to monitor
+instead of multiple.
+
+**Q2: Why not model each device profile update as a firmware update that includes the desired software and configurations as well**
+
+Although this is a more robust approach, it is not feasible on all kinds of devices,
+especially the ones that does not support delta firmware updates.
+On such devices, pushing the entire firmware binary for each iterative change would considerably increase the binary size overhead.
+
+# Device profile capability
+
+A device that supports device profile operation must declare that capability by publishing the following message:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile' '{
+  "name": "prod-profile",
+  "version: "v1"
+}'
+```
+
+If the current profile information is not known upfront, publish an empty JSON (`{}`) instead.
+The firmware information, supported software and configuration types must be declared separately with their respective capability messages.
+
+# Device profile command
+
+Once the `device_profile` capability is declared, the device can receive `device_profile` commands
+by subscribing to `<root>/<device-topic-id>/cmd/device_profile/+` MQTT topics.
+For example, subscribe to the following topic for the `main` device:
+
+```sh te2mqtt formats=v1
+tedge mqtt sub 'te/device/main///cmd/device_profile/+'
+```
+
+A `device_profile` command with `id` "1234" is triggered as follows:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile/1234' '{
+  "status": "init",
+  "name": "prod-profile",
+  "version": "v2",
+  "profile": [
+    {
+      "module": "firmware",
+      "include": true,
+      "name": "core-image-tedge-rauc",
+      "remoteUrl": "https://abc.com/some/firmware/url",
+      "version": "20240430.1139"
+    },
+    {
+      "module": "software",
+      "include": true,
+      "name": "collectd",
+      "version": "latest",
+      "type": "apt",
+      "action": "install"
+    },
```

Comment:
1. I don't think it is a good idea mixing the the `module` property with the other properties, as it makes it more difficult to avoid name clashes between the device profile data structure and the component processing the item (e.g. software_update).
2. Updating of multiple software packages should still be possible when calling the `software_update` operation. It is probably better to just allow the payload within the array of items rather than flattening it out.
3. Using the software operation `software_update` instead of `software`, otherwise we have to maintain a mapping between the device profile component, and the translation to the different sub operations doing the work.

Reply:
Resolved as suggested, by using the same payload format as used by the standalone operations.